src/components/Nav/Nav.tsx

```diff
@@ -196,6 +194,14 @@ export class Nav extends React.Component<INavProps, INavState> implements INav {
     ev.preventDefault();
     ev.stopPropagation();
   }
+
+  private _onNavLinkClicked(link: INavLink, ev: React.MouseEvent): void {
+    this._selectedKey = link.key;
+
+    if (link.onClick) {
+      link.onClick(ev);
```

Comment:
that should be the caller's responsibility for sure. Consider a scenario like: have an href to navigate away, but have an onclick that logs telemetry. In that case, you want default behavior, but you still want the onclick to work.